docs/credentials.md

```diff
@@ -1,250 +1,508 @@
-# Setting Up Credentials and Region
+# <!-- omit in toc --> Setting Credentials and Region
 
-## Credentials
+- [Set credentials](#set-credentials)
+  - [Set credentials for all services with environment variables](#set-credentials-for-all-services-with-environment-variables)
+  - [Get credentials from the AWS shared credentials file](#get-credentials-from-the-aws-shared-credentials-file)
+  - [Set credentials for an individual service](#set-credentials-for-an-individual-service)
+  - [Get credentials from an EC2 or container IAM role](#get-credentials-from-an-ec2-or-container-iam-role)
+  - [Get credentials from a command line process](#get-credentials-from-a-command-line-process)
+  - [Assume an IAM role with credentials from the environment or an IAM role](#assume-an-iam-role-with-credentials-from-the-environment-or-an-iam-role)
+  - [Assume an IAM role with credentials from another profile](#assume-an-iam-role-with-credentials-from-another-profile)
+  - [Use multifactor authentication](#use-multifactor-authentication)
+- [Set region](#set-region)
+  - [Set region for all services with an environment variable](#set-region-for-all-services-with-an-environment-variable)
+  - [Get region from the AWS config file](#get-region-from-the-aws-config-file)
+  - [Set region for an individual service](#set-region-for-an-individual-service)
+- [Set profile](#set-profile)
+- [Reference](#reference)
+  - [Service settings](#service-settings)
+  - [Environment variables](#environment-variables)
+  - [Shared credentials file](#shared-credentials-file)
+  - [Config file](#config-file)
 
-In order to use a Paws package, you must provide it with your
-credentials.
+---
 
-Credentials can be set in the following five ways, and Paws will look
-for them in this order.
 
-1.  Custom service config.
-2.  R environment variables.
-3.  System environment variables (Mac and Linux Only).
-4.  AWS credentials file.
-5.  AWS shared configuration file.
-6.  IAM role.
+# Set credentials
 
-If you are running the package on an instance with an appropriate IAM
-role, Paws will use it automatically and you don’t need to do anything
-extra.
+In order to use AWS APIs, you must provide your credentials. Paws will look for
+your credentials in the following order.
 
-### Setting Credentials with Custom Service Config
+1.  In settings provided to an individual service.
```

Comment:
It is a little confusing that we list the order here:
1) By service
2) By environment variables
3) In files
4) IAM role

but then then the information isn't displayed in that order and instead as:

1) Env variables
2) File
3) By service
4) IAM role
5) Command line
6) Assume Role
7) Multifactor

We should:
* make that ordering consistent
* make these links to the corresponding section
* add the other options to this list

Reply:
I think then we should take that 4 point list out, because it appears again more appropriately in the Reference section and its aim is to explain in what order different sources are checked. Whereas the goal of the first half of the document is to explain how to address a specific use case (how to).

Reply:
I think that is fine. I think we should add a header to that later section, though, so it is easier to find.